### ISSUE TITLE
doxygen gets confused with duplicate class definitions and recursion detection

### ISSUE THREAD

**Issue by PlasmaHH** [doxygen/doxygen#10165]
**Describe the bug**
When two files in the project contain the same two classes deriving in the same way, doxygen issues a "Detected potential recursive class relation between class B and base class A" warning

**Expected behavior**
No such warning

**To Reproduce**
Put the following code into two header files:

```
class A {};
class B : A{};
```

**Version**
1.9.7

**Additional context**
This can happen in real world projects where you have different implementations under the same class hierarchy. This is probably causes by a fix to bug 690787 (#5005) which detects something similar.

It seems to be that the check at doxygen.cpp:4553 i.e.
```
if (!cd->isSubClass(baseClass) && cd!=baseClass && cd->isBaseClass(baseClass,true,templSpec)==0) // check for re        cursion, see bug690787
```
gets confused when the names have been encountered in a previous run already...

**Comment by albert-github** [comment]
Some bisecting, looks like the problem is introduced with:
```
57f5ad207379b49246493850269bc8be49fa95ff is the first bad commit
commit 57f5ad207379b49246493850269bc8be49fa95ff
Date:   Fri Dec 9 19:27:50 2022 +0100

    issue #9715: segmentation fault (recursive classes?)

 src/doxygen.cpp | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)
```
but that also the extra fix:
```
Commit: f502b1932177fec9c7d424f43415c8b861112da2 [f502b19]
Date: Tuesday, December 13, 2022 9:40:40 PM
issue #9720: Recursive class relation warning when using type aliases starting with 57f5ad2
```
does not solve this problem.

**Comment by doxygen** [comment]
@PlasmaHH Can you please verify if the referenced commit fixes the problem for you? Please do not close the issue, as this will be done automatically when the next official release is made available.

**Comment by PlasmaHH** [comment]
@doxygen It fixes the problem for my local testcase. I cannot tell yet if it does for the whole production codebase as I am on vacation and will be back in a bit more than 2 weeks, but from my slightly more complicated testcase condensed from production code that I have with me it looks fine. 

(btw. I noticed that doxygen build from that git commit crashes for me after main exit when built with use_libclang, known issue or worth a bugreport?)

**Comment by albert-github** [comment]
Worth a bug report.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
